### DISCUSSION TITLE
Simple: How to run a cloud function that returns a JSON?

### DISCUSSION THREAD

**Discussion by pktealshift** [netreconlab/Parse-Swift#164] [Q&A] [answered]
I've done this a thousand times in JS, but for the life of me I CANNOT figure out what to declare the ReturnType of my ParseCloud struct so it accepts arbitrary JSONs return values. I need some sample code please.

Closest example in the CloudCode playground pages is:
`typealias ReturnType = [String: Int]`
but I need it to accept stuff like `[String: ParseObject]` (doesn't compile)
or even better, any valid JSON would be useful.

**Comment by pktealshift** [comment]
The problem I'm currently trying to solve is how I can get two objects from one cloud function which returns this JSON:
```
{
   "object1": ParseObjectA,
   "object2": ParseObjectB
}
```

**Reply by jaysonng** [reply, marked as answer]
so the cloud function works fine and returns this JSON?

what error are you specifically getting?

what is the struct of ParseObjectA and ParseObjectB?

seems to me you need to make a new struct that has those values.


``` swift
struct NewObject {

    var object1: ParseObjectA?
    var object2: ParseObjectB?

}
```

which would be your ParseCloudable returnType for example:

```swift
struct NewObjectCloud: ParseCloudable {

    typealias ReturnType = NewObject
    var functionJobName: String = "getObject" // this would be the cloud code function name

    //: Parameters
    var object1ID: String?
    var object2ID: String?
}
```

if everything works ok, you should get back a `NewObject` object.

**Reply by pktealshift** [reply]
Ohhh I see how it works now. Thank you for explaining it so clearly!
